### DISCUSSION TITLE
Should we release 0.72.2?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#76] [Patches]
## Released, please refer to #82 

---

Current Release: [0.72.1](https://github.com/facebook/react-native/releases/tag/v0.72.1)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of issues we're aware of

* https://github.com/facebook/react-native/issues/38006
* https://github.com/facebook/react-native/issues/37871
* https://github.com/facebook/react-native/issues/37944

#### Ordered list of picks (and pick status)

1. https://github.com/facebook/react-native/pull/38142 ✅
1. https://github.com/facebook/react-native/commit/c05d822f7daa92e8af2ec2cd97a9897425624cc2 ✅
2. https://github.com/facebook/react-native/commit/3e3032636dc90a21a499492dcb88f819bcf4f003 ✅
3. https://github.com/facebook/react-native/commit/29f2602ff9c3c9a9999c54a6004c99d6fd15ebc3  ✅
4. https://github.com/facebook/react-native/commit/a4a065549677c61eb91bf587032976ed48c75821 🟡✅
5. https://github.com/facebook/react-native/commit/5bda54c1f183fbc51dc7264b0ab94d5bbcc3f172 ✅
6. https://github.com/facebook/react-native/commit/0c9c57a9f73294414d92428c5d2472dc1e1e5e96 ✅
7. https://github.com/facebook/react-native/commit/41477c898cf5726eae9edbb1596366a6eea2b01e ✅
8. https://github.com/facebook/react-native/commit/58adc5e4b9ab74b67b4af04d1e72c387af848ea7 ✅
9. https://github.com/facebook/react-native/commit/86b5b4cc34dfa8b8ec1852f2cd195361e1a9311d ✅
10. https://github.com/facebook/react-native/commit/28f4ebab8ab4b0f337699e6a135e2aa983866f42 ✅
11. https://github.com/facebook/react-native/commit/0ccbd65581304faa286b452f75058b6292a6240f ✅
12. https://github.com/facebook/react-native/commit/8d2eec367dd6fbd60792ca1bde12b875a8261fa6 ✅
13. https://github.com/facebook/react-native/commit/eaafc260c14674bfbf7ce3a647404c70d546854f ✅

#### Not merged at the time of picking

1. https://github.com/facebook/react-native/pull/35803

#### Local commits to backport to main

1. https://github.com/facebook/react-native/commit/8ce471e2fa802cc50ff2d6ab346627cb5f6d79b4 (from 72.1)

**Comment by efstathiosntonas** [comment]
Thank you guys for all your hard work.

@kelset on release notes the correct SocketRocket version is 0.6.1, there’s a typo on the PR title and it got through to the changelog, just fyi.

**Reply by kelset** [reply]
thanks, fixed

**Comment by adrianha** [comment]
hi, can we cherry pick https://github.com/facebook/react-native/commit/3e3032636dc90a21a499492dcb88f819bcf4f003? 🙏

**Comment by efstathiosntonas** [comment]
react-native-upgrade helper hasn't picked up 0.72.1 yet

**Reply by kelset** [reply]
on it - it's my bad, because of how the script works it picked up a local change I had and added to the other changes

**Comment by efstathiosntonas** [comment]
After upgrading to 0.72.0 (and now on 0.72.1) the `hermes` pod is not being updated to match the react-native version after the `pod install`. Don't know if this is a `pods`/`.podspec` issue (or my setup issue) but I have to run `pod update` for it to pick it up.

**Reply by cipolleschi** [reply]
I'll try to give this a shot today.

Do you happen to have the error that is reported? (BTW, we didn't bump hermes, there were no changes in it between 0.72.0 and 0.72.1, so even using the old version would not be a problem! 😉  )

**Reply by efstathiosntonas** [reply]
I get no error, it just won't install when I run `pod install` after upgrading react native. It gets synced with rn version only on `pod update`. 

~edit: what will happen if hermes is bumped and `pod install` fails to install / align with rn? This could confuse devs since they expect to be aligned. Ofcourse we should check the pod install output but it might slip.~

**Reply by cipolleschi** [reply]
Ah, that's expected, then. 
Looking [here](https://github.com/facebook/react-native/blob/0.72-stable/packages/react-native/scripts/cocoapods/jsengine.rb#L38), hermes is automatically updated when the `tag` changes. As I said, we didn't bump Hermes as there were no changes there, so the `tag` didn't change. And Cocoapods was smart enough to reuse the same version you already had, as it is actually the same! ;)

**Reply by efstathiosntonas** [reply]
copy! thanks for the heads up @cipolleschi

**Comment by efstathiosntonas** [comment]
Can we pick this one: 

- https://github.com/facebook/react-native/pull/37958

it fixes this: https://github.com/facebook/react-native/pull/36930#issuecomment-1593028467 and ~probably~ this: https://github.com/facebook/react-native/issues/37991

**Comment by sammy-SC** [comment]
Can we pick: 
- https://github.com/facebook/react-native/commit/5bda54c1f183fbc51dc7264b0ab94d5bbcc3f172

It disables one optimisation in Fabric which isn't critical but introduced a bug.

**Comment by lunaleaps** [comment]
Merge https://github.com/facebook/react-native/pull/38142 
to fix Android regression of remote debugging mentioned in https://github.com/facebook/react-native/issues/38006

Note: this does not fix iOS remote debugging, still looking into that

**Reply by mikehardy** [reply]
react-native-firebase still uses remote debugging as it's primary conduit for e2e testing - we have a huge test library that we haven't migrated to another style yet.

So I really, personally, appreciate this one. Huge thanks to @javache for the fix and @lunaleaps for carrying it over!

**Reply by lunaleaps** [reply]
We should also pick: https://github.com/facebook/react-native/commit/41477c898cf5726eae9edbb1596366a6eea2b01e for https://github.com/facebook/react-native/issues/38204

**Comment by louiszawadzki** [comment]
Hi! 
Can we cherry pick this fix: https://github.com/facebook/react-native/pull/38121? (commit: https://github.com/facebook/react-native/commit/a4a065549677c61eb91bf587032976ed48c75821)
It enables turbo modules with Swift code to work on 0.72 with the new architecture.

**Comment by cortinico** [comment]
I would like to have included this commit:
- https://github.com/facebook/react-native/commit/86b5b4cc34dfa8b8ec1852f2cd195361e1a9311d

Inside .2 as I believe this could mitigate this issue:
- https://github.com/facebook/react-native/issues/36296

If that is confirmed as the proper fix, we can backport it to previous versions as well

**Reply by Kudo** [reply]
also this to fix the hermes inspector on android please: https://github.com/facebook/react-native/commit/41477c898cf5726eae9edbb1596366a6eea2b01e

**Reply by efstathiosntonas** [reply]
@cortinico this needs to be built from source or we can patch it and test it?

**Reply by cortinico** [reply]
It needs to be built from source as it's part of the Hermes engine build logic

**Comment by cipolleschi** [comment]
We forgot to pick this:
- https://github.com/facebook/react-native/commit/c05d822f7daa92e8af2ec2cd97a9897425624cc2

Which is already in 0.70 and 0.71

**Comment by Kudo** [comment]
please help to pick this commit for the `@react-native/metro-config`: https://github.com/facebook/react-native/commit/0ccbd65581304faa286b452f75058b6292a6240f

**Comment by scrgh** [comment]
Any update on this (0.72.2) release ?

**Reply by cipolleschi** [reply]
Please, don't ask ETAs in this discussion. 
The release crew will release a new patch when there are enough commit and the chunk of work required to fix identified bugs is done.
Thank you so much!

**Comment by Titozzz** [comment]
Hello we probably want to cherry this one: https://github.com/facebook/react-native/commit/28f4ebab8ab4b0f337699e6a135e2aa983866f42

**Comment by cipolleschi** [comment]
Forgot to mention this:
* https://github.com/facebook/react-native/commit/8d2eec367dd6fbd60792ca1bde12b875a8261fa6

It landed on Friday!

**Comment by oblador** [comment]
I'm nominating this metro fix that solves a regression in `type: module` project: 

* https://github.com/facebook/metro/commit/692293339e3b9f542f7b83c04bdccf46fef05766

Not sure of how the release flow works for metro, but I'm guessing the fix needs to be back ported to the 0.76 branch and released as a patch. cc @huntie

**Reply by kelset** [reply]
we're literally releasing 0.72.2 right now, but let's try to get it into the next patch 😅

**Reply by oblador** [reply]
OK! I think it might be enough to just do the metro patch release actually 👍
